instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import Comment from './Comment';
+import './CommentSection.css';
+
+CommentSection.propTypes = {
+    postData: PropTypes.shape({
+        comments: PropTypes.array,
+        timestamp: PropTypes.string
+    })
+};
+
+class CommentSection extends React.Component {
+
+    state = {
+        comments: this.props.post.comments,
+        username: ""
+    };
+
+    componentDidMount() {
+        const id = this.props.post.imageUrl;
+        this.addNewComment();
+    }
+
+    addNewComment = event => {
+        event.preventDefault();
+        if (this.state.inputText) {
+            this.setState({
+                users: this.state.users,
```

Comment:
The same is happening here. You're trying to set a property `user` on state but there is not `user` property on the state object. For addNewComment, the function should only manage the logic for adding a new comment. You DO have a comments property on state. 👍